wallet/src/transaction/dump.rs

```diff
@@ -1,112 +1,157 @@
-use crate::{transaction::WitnessBuilder, Settings};
-use chain_addr::Address;
+use crate::store::UtxoStore;
+
+use super::builder::{AddInputStatus, TransactionBuilder};
+use super::witness_builder::{OldUtxoWitnessBuilder, UtxoWitnessBuilder, WitnessBuilder};
 use chain_impl_mockchain::{
-    fee::FeeAlgorithm as _,
-    fragment::{Fragment, FragmentRaw},
-    transaction::{Input, Output, TxBuilderState},
-    value::Value,
+    fragment::Fragment,
+    transaction::{Balance, Input, NoExtra, Output, Transaction},
 };
 
-/// Dump all the values into transactions to fill one address
-pub struct Dump {
-    settings: Settings,
-    address: Address,
-    inputs: Vec<Input>,
-    witness_builders: Vec<WitnessBuilder>,
-    outputs: Vec<(Vec<Input>, FragmentRaw)>,
-    ignored: Vec<Input>,
+pub struct DumpIter<'a, W> {
+    settings: &'a crate::Settings,
+    address: &'a chain_addr::Address,
+    wallet: &'a mut W,
 }
 
-impl Dump {
-    /// dump all the associated input to the given address
-    pub fn new(settings: Settings, address: Address) -> Self {
-        Self {
-            settings,
-            address,
-            inputs: Vec::with_capacity(255),
-            witness_builders: Vec::with_capacity(255),
-            outputs: Vec::new(),
-            ignored: Vec::new(),
+pub type DumpDaedalus<'a> = DumpIter<'a, crate::scheme::rindex::Wallet>;
+pub type DumpIcarus<'a> =
+    DumpIter<'a, crate::scheme::bip44::Wallet<chain_impl_mockchain::legacy::OldAddress>>;
+pub type DumpFreeKeys<'a> = DumpIter<'a, crate::scheme::freeutxo::Wallet>;
+
+pub fn send_to_one_address<S: Clone, K: 'static, WB: WitnessBuilder>(
+    settings: &crate::Settings,
+    address: &chain_addr::Address,
+    utxo_store: &UtxoStore<S, K>,
+    mk_witness: &'static dyn Fn(hdkeygen::Key<S, K>) -> WB,
+) -> Option<(Transaction<NoExtra>, Vec<Input>)> {
+    let payload = chain_impl_mockchain::transaction::NoExtra;
+
+    let mut builder = TransactionBuilder::new(settings, payload);
+    let utxos = utxo_store.utxos();
+
+    let mut ignored = vec![];
+
+    // TODO: return this?
```

Comment:
no, but you need to check the fragment produced though, and you already doing it in the iterator implementations so it's okay